libraries/AP_Compass/AP_Compass.h

```diff
@@ -443,6 +453,8 @@ friend class AP_Compass_Backend;
     struct mag_state {
         AP_Int8     external;
         bool        healthy;
+        bool        registered;
+        Compass::Priority priority;
```

Comment:
It's perhaps error prone to have a variable called "priority" and a TSIndex object called "Priority"

Reply:
"Priority" is not a TSIndex object, but a typedef that is created using template class TSIndex. its a type that is used to keep distinction between different types of indexes. And make it compile time strict.